## Desafío 1

## Desafío 1

In [118]:
import pandas as pd
import numpy as np
df = pd.read_csv('worldcup2014.csv')
df.head(5)

team     continent group  group_pos  cantidad_juegos  \
0       Brazil  southamerica     A          1                3   
1       Mexico  northamerica     A          2                3   
2      Croatia        europe     A          3                3   
3     Cameroon        africa     A          4                3   
4  Netherlands        europe     B          1                3   

   juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0               2                 1                0            7   
1               2                 1                0            4   
2               1                 0                2            6   
3               0                 0                3            1   
4               3                 0                0           10   

   goles_contra goles_diferencia  puntos  clasificado  
0             2                5       7            1  
1             1                3       7            1  
2             6                0       3            0  
3             9               −8       0            0  
4             3                7       9            1

## Desafío 2

In [119]:
df['continent'].value_counts(normalize = True)

europe          0.40625
southamerica    0.18750
africa          0.15625
northamerica    0.12500
asia            0.12500
Name: continent, dtype: float64

#### El continente con una mayor presencia es: europe
#### La probabilidad de elegir un equipo asiático al azar es: 0.125
#### La probabilidad de elegir un equipo africano al azar es: 0.156

## Desafío 3

In [120]:
europe_df = df[df['continent'] == "europe"]
southamerica_df = df[df['continent'] == "southamerica"]
africa_df = df[df['continent'] == "africa"]
northamerica_df = df[df['continent'] == "northamerica"]
asia_df = df[df['continent'] == "asia"]

## Desafío 4

In [121]:
df_array = [europe_df, southamerica_df, africa_df, northamerica_df, asia_df]
for data in df_array:
    continent = data['continent'].unique()[0]
    clasificados = data['clasificado'].value_counts(normalize = True).get(1)
    print("La probabilidad de pasar a la siguiente ronda en {} es {}".format(continent, clasificados))

La probabilidad de pasar a la siguiente ronda en europe es 0.46153846153846156
La probabilidad de pasar a la siguiente ronda en southamerica es 0.8333333333333334
La probabilidad de pasar a la siguiente ronda en africa es 0.4
La probabilidad de pasar a la siguiente ronda en northamerica es 0.75
La probabilidad de pasar a la siguiente ronda en asia es None


#### La probabilidad de que un país asiático pase a la siguiente ronda es 0
#### Los dos continentes que tuvieron la mayor probabilidad de clasificar son southamerica y northamerica
#### La probabilidad de no clasificar un país europeo es 0.54

## Desafío 5

In [122]:
for d in [africa_df, europe_df, asia_df, northamerica_df, southamerica_df]:
    juegos_ganados = pd.Series(np.where(d['juegos_ganados'] == 0, 0, 1))
    print(juegos_ganados.value_counts(normalize = True))

1    0.6
0    0.4
dtype: float64
1    0.846154
0    0.153846
dtype: float64
0    1.0
dtype: float64
1    0.75
0    0.25
dtype: float64
1    1.0
dtype: float64


#### El continente que tuvo una mayor probabilidad de ganar juegos fue southamerica
#### El continente que presentó un nivel similar entre juegos ganados y perdidos fue africa

In [123]:
for d in [africa_df, europe_df, asia_df, northamerica_df, southamerica_df]:
    juegos_perdidos = pd.Series(np.where(d['juegos_perdidos'] == 0, 0, 1))
    print(juegos_perdidos.value_counts(normalize = True))

1    1.0
dtype: float64
1    0.692308
0    0.307692
dtype: float64
1    1.0
dtype: float64
1    0.5
0    0.5
dtype: float64
1    0.5
0    0.5
dtype: float64


#### El continente que tuvo una mayor probabilidad de perder juegos fue africa y asia